Statsman/Core/Generators/Services/StatisticsTimeService.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+using Watchman.Common.Models;
+using Watchman.DomainModel.Messages;
+
+namespace Statsman.Core.Generators.Services
+{
+    public class StatisticsTimeService
+    {
+        public IEnumerable<TimeRange> GetTimeRangeMovePerPeriod(string period, DateTime oldestMessageDatetime) //TODO test
+        {
+            var startOfCurrentPeriod = this.GetStartOfCurrentPeriod(period);
+            var moveForward = this.GetMoveForward(period);
+            var moveBackward = this.GetMoveBackward(period);
+            var daysAtEnd = this.GetDaysAtEnd(period);
+
+            var periodTimeRange = TimeRange.Create(startOfCurrentPeriod, startOfCurrentPeriod.AddDays(moveForward.Invoke(startOfCurrentPeriod) - daysAtEnd).AddMilliseconds(-1));
+            periodTimeRange = periodTimeRange.Move(TimeSpan.FromDays(-moveBackward.Invoke(startOfCurrentPeriod)));
+            var iterableTimeRange = periodTimeRange.MoveWhile(x => !x.Contains(oldestMessageDatetime), x => TimeSpan.FromDays(-moveBackward.Invoke(x.Start)));
+            return iterableTimeRange;
```

Comment:
```suggestion
            return TimeRange
                .Create(startOfCurrentPeriod, startOfCurrentPeriod.AddDays(moveForward.Invoke(startOfCurrentPeriod) - daysAtEnd).AddMilliseconds(-1))
                .Move(TimeSpan.FromDays(-moveBackward.Invoke(startOfCurrentPeriod)))
                .MoveWhile(x => !x.Contains(oldestMessageDatetime), x => TimeSpan.FromDays(-moveBackward.Invoke(x.Start)));
```
ja bym to zrobił mniej więcej tak.